### ISSUE TITLE
Migrate away from openbox and compton

### ISSUE THREAD

**Issue by agaida** [lxqt/lxqt#1011]
hi, i would like to suggest to move/migrate away from openbox/compton to xfwm4 as our recommend WM. Reasons: openbox as lowest common nominator don't make sense anymore, xfwm4 is available for all major linux and bsd distributions. compton is stalled and out of date in the most distributions. i would go a step further and say it is dead. Both are not ready for wayland. I know that xfwm4 isn't too but there are more chances that xfwm4 will be ready for Wayland. For the time beeing xfwm4 is the far more capable WM without the need for packages like obconf-qt and compton-conf, so it would ease our development.

**Comment by jleclanche** [comment]
What are the advantages of xfwm4 over openbox?

**Comment by Esclapion** [comment]
xfwm4 depends on gtk2 and gtk3, not qt5 ?

**Comment by tsujan** [comment]
> openbox as lowest common nominator don't make sense anymore

In which sense doesn't it make sense when it's so configurable?

> compton is stalled and out of date in the most distributions

That *may* be true but it still works without problem here.

**Comment by Vladimir-csp** [comment]
Openbox is feature-complete, stable, lightweight, does not depend on whole giant toolkit. There is just nothing more to develop in it, but it is mainained. And it also has some killer features like the way it handles keyboard shortcuts and advanced window management actions. Compton is now pretty stable. Both are extremely lightweight.
If not this whole Wayland thing, they could go on forever. Until anything remotely comparable emerges, Openbox is the best choice on X.
Xfwm4 depends on GTK2 and xfce components like xfconf - an ananlog of GNOME's gconf/dconf, which is an analog of you-know-whose registry. I don't think it is worth to depend on such horrors. :)

**Comment by sixsixfive** [comment]
>What are the advantages of xfwm4 over openbox?

probably wayland support also with a possible gtk3 port comes scaling/hidpi support since openbox has some great disadvantages on hidpi(max button size is ~14px? wich is ~7px on hidpi)

there are some alternatives:

* https://github.com/xiangzhai/eggwm/
* https://sourceforge.net/projects/jwm 
* fluxbox is also always nice

**Comment by Vladimir-csp** [comment]
I would suggest creating LXQt module that would work similar to wmctrl - interact with a EWMH/NetWM compatible X Window Manager (or use wmctrl itself) for governing windows in response to events or keyboard shortcuts.
This would decouple LXQt from specific WM, so that any compatible WM with blanked out shortcut configuration would fit right in with minimal tweaks.

**Comment by manjarqo** [comment]
It is good idea, as for me, i have removed openbox and obconf and install xfwm4 it lightweight too, my system eating 250 Mb memory, and [looks great ](http://imgur.com/w0NhfvO)

**Comment by agaida** [comment]
@Vladimir-csp: sounds good
@all: one could consider the project compton dead - no release for few years and the last commit end of 2015. The bugtracker is not about solving bugs but about how great compton is - not the best base to build a modern desktop environment. 

I mentioned xfwm4 because it it simply work and is available in nearly all distributions including *bsd, it provides edge snapping, a compositor and other nice features. It would be not the best possible solution, but a good lowest common denominator.

**Comment by jleclanche** [comment]
The plan honestly has always been to move to kwin long term, with the goal in mind to make kwin more lightweight, less kde-dependent etc (a situation which has improved greatly since we started the project).

Keep in mind we want to be compatible with wayland and there's not many WMs that offer that right now. Kwin and Sway are the only two reliable ones (and as nice as sway is, it's not what I would recommend as a default WM).

**Comment by tsujan** [comment]
Although I like Openbox+Compton, yesterday I felt nostalgic for Compiz and hopelessly tried to compile it with Arch patches under Debian Testing. Much to my surprise, not only the compilation finished successfully but also Compiz worked smoothly with LXQt -- almost like old days with Gnome2! Viva Arch community!

While kwin created coredump on logging out, there's no Compiz coredump here! I just disabled Compton, put `compiz --replace` in `Session Settings → Window Manager` and logged out and in :)

I may even stay with it until Wayland comes.

**Comment by physkets** [comment]
Has anyone looked at [Orbital](https://github.com/giucam/orbital) as an option? How does it compare to Kwin?

**Comment by palinek** [comment]
> Has anyone looked at Orbital

I've just tried to build/test it.... but I wasn't able to go further than the "splash screen" :(

![screenshot](https://cloud.githubusercontent.com/assets/10990929/15647066/a38218ac-2661-11e6-9a45-d0a028cf7baa.png)

**Comment by paulolieuthier** [comment]
Orbital looks like a nice project to keep an eye on.

**Comment by tsujan** [comment]
@palinek Is Wayland stable on Debian?

**Comment by palinek** [comment]
> Is Wayland stable on Debian?

I'm not using wayland :)... I just built orbital and gave it a try under the X.

**Comment by f2404** [comment]
@palinek Hmm, should it run under X?

> Orbital is a Wayland compositor and shell

**Comment by palinek** [comment]
> Hmm, should it run under X?

Based on it's readme, it should:
> Running Orbital
---

> Now you can just run orbital, to run it if you are inside an X or a Wayland session. To start its own dedicated session run orbital-launch from a tty.

**Comment by tsujan** [comment]
After re-experiencing Compiz, I won't be satisfied by less than it anymore. The best Wayland-based candidate will be KWin if it keeps its effects. Current (and even a bit older) graphics cards are wasted by simple WMs, IMO.

**Comment by agaida** [comment]
wayland stable :)  - there is no such thing right now. But one could test it in debian sid i thing - runs more or less fine with gnome and kde, but still have glitches.

**Comment by giucam** [comment]
@palinek The console output likely has a clue as to why it stops at the splash screen.

**Comment by jubalh** [comment]
The Lumina Desktop guys plan to (already started?) developing a new WM. Just throwing this into the discussion ans maybe joining forces there would be an option.

**Comment by palinek** [comment]
> The console output likely has a clue as to why it stops at the splash screen.

@giucam Here is the output:
```
$ ~/opt/weston/bin/orbital
Loading module '/home/palco/opt/weston/lib/libweston-0/x11-backend.so'
Loading module '/home/palco/opt/weston/lib/libweston-0/gl-renderer.so'
EGL client extensions: EGL_EXT_client_extensions EGL_EXT_platform_base
               EGL_EXT_platform_wayland EGL_EXT_platform_x11
               EGL_KHR_client_get_all_proc_addresses EGL_MESA_platform_gbm
warning: EGL_EXT_buffer_age not supported. Performance could be affected.
warning: EGL_EXT_swap_buffers_with_damage not supported. Performance could be affected.
Using gl renderer
EGL version: 1.4 (DRI2)
EGL vendor: Mesa Project
EGL client APIs: OpenGL OpenGL_ES OpenGL_ES2 OpenGL_ES3 
EGL extensions: EGL_CHROMIUM_sync_control EGL_EXT_create_context_robustness
               EGL_EXT_image_dma_buf_import EGL_KHR_create_context
               EGL_KHR_fence_sync EGL_KHR_get_all_proc_addresses
               EGL_KHR_gl_renderbuffer_image EGL_KHR_gl_texture_2D_image
               EGL_KHR_gl_texture_cubemap_image EGL_KHR_image
               EGL_KHR_image_base EGL_KHR_image_pixmap
               EGL_KHR_surfaceless_context EGL_KHR_wait_sync
               EGL_MESA_configless_context EGL_MESA_drm_image
               EGL_MESA_image_dma_buf_export EGL_NOK_swap_region
               EGL_NOK_texture_from_pixmap EGL_NV_post_sub_buffer
               EGL_WL_bind_wayland_display
GL version: OpenGL ES 3.1 Mesa 11.2.2
GLSL version: OpenGL ES GLSL ES 3.10
GL vendor: Intel Open Source Technology Center
GL renderer: Mesa DRI Intel(R) HD Graphics 5500 (Broadwell GT2) 
GL extensions: GL_EXT_blend_minmax GL_EXT_multi_draw_arrays
               GL_EXT_texture_filter_anisotropic
               GL_EXT_texture_compression_dxt1 GL_EXT_texture_format_BGRA8888
               GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth24
               GL_OES_element_index_uint GL_OES_fbo_render_mipmap
               GL_OES_mapbuffer GL_OES_rgb8_rgba8 GL_OES_standard_derivatives
               GL_OES_stencil8 GL_OES_texture_3D GL_OES_texture_float
               GL_OES_texture_float_linear GL_OES_texture_half_float
               GL_OES_texture_half_float_linear GL_OES_texture_npot
               GL_OES_EGL_image GL_OES_depth_texture
               GL_OES_packed_depth_stencil GL_EXT_texture_type_2_10_10_10_REV
               GL_OES_get_program_binary GL_APPLE_texture_max_level
               GL_EXT_discard_framebuffer GL_EXT_read_format_bgra
               GL_NV_fbo_color_attachments GL_OES_EGL_image_external
               GL_OES_EGL_sync GL_OES_vertex_array_object
               GL_ANGLE_texture_compression_dxt3
               GL_ANGLE_texture_compression_dxt5 GL_EXT_texture_rg
               GL_EXT_unpack_subimage GL_NV_draw_buffers GL_NV_read_buffer
               GL_NV_read_depth GL_NV_read_depth_stencil GL_NV_read_stencil
               GL_EXT_draw_buffers GL_EXT_map_buffer_range GL_KHR_debug
               GL_OES_depth_texture_cube_map GL_OES_surfaceless_context
               GL_EXT_color_buffer_float GL_EXT_separate_shader_objects
               GL_EXT_shader_integer_mix GL_EXT_draw_elements_base_vertex
               GL_KHR_context_flush_control GL_OES_draw_elements_base_vertex
               GL_OES_texture_storage_multisample_2d_array
               GL_EXT_blend_func_extended GL_EXT_buffer_storage
               GL_EXT_shader_samples_identical
GL ES 2 renderer features:
               read-back format: BGRA
               wl_shm sub-image to texture: yes
               EGL Wayland extension: yes
Chosen EGL config details:
               RGBA bits: 8 8 8 0
               swap interval range: 0 - 1000
x11 output 800x400, window id 75497477
Using 'wayland-0'
Launching '/home/palco/opt/weston/libexec/orbital-authorizer-helper'...
xserver listening on display :1
Launching '/home/palco/opt/weston/libexec/orbital-splash'...
Launching '/home/palco/opt/weston/libexec/startorbital'...
autostart!!
populate /home/palco/.config/autostart
Autostarting /home/palco/.config/autostart/set_mouse_precision.desktop: 'mouse_precision.sh'
Autostarting /home/palco/.config/autostart/set_video.desktop: 'video.sh'
forked X server, pid 6459
glamor: EGL version 1.4 (DRI2):
[D orbital-client - /home/palco/work/oss/orbital/src/client/client.cpp:108] == Wayland display socket: 48
[D orbital-splash - unknown:0] == Using Wayland-EGL
[W orbital-client - /home/palco/work/oss/orbital/src/client/style.cpp:51] == Could not find the style "chiaro"
[D orbital-client - /home/palco/work/oss/orbital/src/client/shellui.cpp:150] == screen "X1"
[W orbital-client - /home/palco/work/oss/orbital/src/client/element.cpp:387] == "Could not find the element 'background'. Check your configuration or your setup."
[W orbital-client - /home/palco/work/oss/orbital/src/client/element.cpp:387] == "Could not find the element 'panel'. Check your configuration or your setup."
[W orbital-client - /home/palco/work/oss/orbital/src/client/element.cpp:387] == "Could not find the element 'overlay'. Check your configuration or your setup."
[W orbital-client - /home/palco/work/oss/orbital/src/client/element.cpp:387] == "Could not find the element 'lockscreen'. Check your configuration or your setup."
[D orbital-client - /home/palco/work/oss/orbital/src/client/client.cpp:235] == Elements for screen "X1" loaded after 0 ms
xfixes version: 5.0
created wm, root 624
```
... and after closing the window
```
[D Orbital - /home/palco/work/oss/orbital/src/compositor/compositor.cpp:545] == Orbital exiting...
xserver exited, code 9
[W Orbital - unknown:0] == QProcess: Destroyed while process ("mouse_precision.sh") is still running.
[W Orbital - unknown:0] == QProcess: Destroyed while process ("video.sh") is still running.
Segmentation fault
[W orbital-client - unknown:0] == The Wayland connection broke. Did the Wayland compositor die?
[W orbital-splash - unknown:0] == The Wayland connection broke. Did the Wayland compositor die?
```

Note: the video.sh/mouse_precision.sh are simple bash scripts (with xinput/xrandr)

**Comment by giucam** [comment]
@palinek Well, it says there: "Could not find the element 'background'. Check your configuration or your setup."
Judging by the path, did you run make install with DESTDIR? That won't work, set the right prefix to cmake if you want to install it in your home.

**Comment by palinek** [comment]
> Judging by the path, did you run make install with DESTDIR? That won't work, set the right prefix to cmake if you want to install it in your home.

Nope, I built it with `cmake -DCMAKE_PREFIX_PATH=$HOME/opt/weston -DCMAKE_INSTALL_PREFIX=$HOME/opt/weston ..'

... oh I almost forgot, I also made changes in the sources for the ETC dirs:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32e9792..882a5cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,4 +31,4 @@ add_custom_target(uninstall
 add_subdirectory(scripts)
 add_subdirectory(data)
 
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/orbital-launcher.desktop DESTINATION /etc/xdg/autostart)
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/orbital-launcher.desktop DESTINATION etc/xdg/autostart)
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index b7585e0..8df1063 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -1,2 +1,2 @@
 configure_file(restricted_interfaces.conf.in ${CMAKE_CURRENT_BINARY_DIR}/restricted_interfaces.conf @ONLY)
-install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/restricted_interfaces.conf DESTINATION /etc/orbital)
+install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/restricted_interfaces.conf DESTINATION etc/orbital)
diff --git a/src/authorizer_helper/main.cpp b/src/authorizer_helper/main.cpp
index fdaeb9e..f8d6aec 100644
--- a/src/authorizer_helper/main.cpp
+++ b/src/authorizer_helper/main.cpp
@@ -88,7 +88,7 @@ public:
         QString path = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation);
         Result res = readFile(path + QLatin1String("/orbital/restricted_interfaces.conf"), global, executable);
         if (res == Result::Unknown) {
-            res = readFile(QStringLiteral("/etc/orbital/restricted_interfaces.conf"), global, executable);
+            res = readFile(QStringLiteral("/home/palco/opt/weston/etc/orbital/restricted_interfaces.conf"), global, executable);
         }
         return res == Result::Allow;
     }
diff --git a/src/compositor/shell.cpp b/src/compositor/shell.cpp
index 4cbd98c..7aedabe 100644
--- a/src/compositor/shell.cpp
+++ b/src/compositor/shell.cpp
@@ -305,7 +305,7 @@ fmt::print(stderr, "autostart!!\n");
             return false;
         });
     } else {
-        populateAutostartList(files, "/etc/xdg/autostart");
+        populateAutostartList(files, "/home/palco/opt/weston/etc/xdg/autostart");
     }
 
     QDir outputDir = QDir::temp();
```

**Comment by giucam** [comment]
Right, it looks in the dirs returned by QStandardPaths for DataLocation, as defined here: http://doc.qt.io/qt-5/qstandardpaths.html#StandardLocation-enum
So that won't include your custom path. In the meanwhile you can move share/orbital/ in ~/.local/share, that shoule make it work.

**Comment by palinek** [comment]
@giucam thanks... with some tweaking in ~/.local/share it is running now

**Comment by paulolieuthier** [comment]
@palinek what do you think of it?

**Comment by palinek** [comment]
> what do you think of it?

Just tested it and tried to run the lxqt-panel in there -> issues discovered right away:
- panel does not place itself correctly
- taskbar doesn't show anything

Here are some glues from @mgraesslin (thanks again):
```irc
(13:16:01) palinek: Hi. A (simple) question: please, what do I need make kwnidowsystem usable under wayland? Is there some needed plugin package?
(13:16:18) palinek: mgraesslin: ^
(13:16:25) bshah: I believe so
(13:16:36) bshah: https://quickgit.kde.org/?p=kwayland-integration.git have kwindowsystem plugin
(13:16:52) mgraesslin: palinek: depends on what you define as "usable" there is a dummy implementation
(13:17:53) palinek: with dummy implementation you mean a "noop" methods (i think i've seen something like that a while ago)?
(13:18:23) mgraesslin: palinek: it's an implementation which matches what Wayland standard protocols allow to use
(13:18:42) mgraesslin: palinek: most of KWindowSystem just doesn't match Wayland - e.g. there are no window ids
(13:22:25) mgraesslin: palinek: so what do you actually need from KWindowSystem on Wayland?
(13:23:03) palinek: looking into the sources... (windowsystem.cpp) WindowSystem::setExtendedStrut
(13:23:03) palinek: mgraesslin: can you give me a clue how do it in wayland
(13:23:19) mgraesslin: not at all, there is nothing like that in the protocol
(13:23:29) palinek: mgraesslin: asking because of LXQt
(13:23:57) mgraesslin: palinek: what we have for that in KWayland is PlasmaShell: you can mark a surface as a panel than struts are applied automatically
(13:24:50) mgraesslin: palinek: see https://api.kde.org/frameworks/kwayland/html/classKWayland_1_1Client_1_1PlasmaShellSurface.html
(13:25:04) palinek: hm... do I need something wayland specific or is that covered by any framework (kwindowsystem, qt)?
(13:25:23) mgraesslin: palinek: but that only works with Compositors which support that protocol which at the moment is probably only KWin
(13:25:33) palinek: :)
(13:25:59) mgraesslin: I think you would need to use KWayland directly
(13:26:05) mgraesslin: let me look for code example
(13:26:57) mgraesslin: palinek: the name Plasma in that doesn't mean that it's only meant for Plasma, we had LXQt in mind when we designed the protcols
(13:27:10) palinek: the KWAYLAND is a tier 1 framework?
(13:27:42) palinek: ...found... good
(13:28:11) mgraesslin: yes
(13:28:14) mgraesslin: https://quickgit.kde.org/?p=plasma-workspace.git&a=blob&h=3910aabe9a8317aa371fc9138821147cf4ca4d99&hb=0d0c4f4a2247e8213deb232d2d80f12480dc05aa&f=shell%2Fpanelview.cpp
(13:28:33) mgraesslin: PanelView shows how to make it a "Panel"
(13:28:45) mgraesslin: the interesting code is in the event handler for QPlatformSurfaceEvent
(13:29:04) mgraesslin: and in PanelView::integrateScreen
(13:29:17) palinek: looking on that
(13:29:19) mgraesslin: what's nice about PlasmaSurface is that it allows to set an absolute position
(13:29:26) mgraesslin: which is something a panel rather wants
(13:29:42) mgraesslin: KWin fully trusts a PlasmaSurface to do the right things
(13:29:54) mgraesslin: and we destroyed the complete security thoughts of Wayland for it ;-)
(13:30:49) mgraesslin: palinek: and for task managers you want to have a look at PlasmaWindow and PlasmaWindowModel
(13:31:07) mgraesslin: (or just use the new libtaskmanager just being under development by Sho_)
```

=> so right now, we have only the kwin as an option for wayland (if we want still stick with kwindowsystem)

**Comment by palinek** [comment]
I forgot to mention that orbital is not a plain window manager, it covers some functionality of panel (it is based on QtQuick)

**Comment by giucam** [comment]
Well, orbital is the compositor/WM, orbital-client is the thing drawing the panel and stuff. Orbital doesn't strictly depend on orbital-client but that's hardcoded for now.

**Comment by giucam** [comment]
On the other hand, you'd probably be better making a new tailored compositor using the libweston library, which is right now becoming a thing.

**Comment by paulolieuthier** [comment]
KWin has its own protocols for things Wayland does not define yet. It's cool so hear that KDE folks had LXQt in mind, but I don't think we should invest time in Wayland right now if it has so many protocol holes (which doesn't mean someone could make a KWin-specific prototype).

Anyway, I created a [TODO page for Wayland](https://github.com/lxde/lxqt/wiki/TODO-for-Wayland) in our wiki. If anyone can add more details there it'll be useful.

**Comment by giucam** [comment]
@paulolieuthier Those kinds of protocols will never be part of core protocol. They are needed for desktop environment, so if desktop environments want to see a standard arise they should come together and make it so. It will never happen from the proper wayland project.
So saying "we'll wait for them" means "we'll wait forever".

**Comment by mgraesslin** [comment]
Sharing some thoughts from KWin/KDE side. As @giucam mentioned we don't expect that this will ever go into Wayland core or even wayland-protocols. Most DEs going to Wayland have a desktop shell in window manager architecture (e.g. Mutter/Gnome Shell). Plasma and LXQt are rather the exception nowadays to have a multi-process architecture. We still believe in the multi-process architecture and the "do one thing and do it right" approach. Given that we decided in the early phases of the Wayland porting that we continue to separate the concerns and use dedicated protocols for the various aspects (unless it violates security concerns - e.g. lock screen and global shortcut handling).

When we started to look at how we do the protocols we looked around at who could be interested in them and thought "LXQt". We agreed that we want to have our protocols in ways that we can support more than Plasma, that we keep the protocols stable, that KWayland becomes a tier1 framework (only Qt dep), etc. We think that the possibility to integrate multiple shells but also to exchange compositors is also important in a Wayland world.

Obviously I would recommend to use KWin. But I would also recommend to look at what various options can offer you and how much work you need to get something running. KWin can shine in that regard as it makes a multi-process architecture and has many protocols already defined for that. E.g.:

- Output configuration - already fully integrated in KScreen (KWin side still under development)
- "DPMS" (turn screen off) - already fully integrated in Plasma's powerdevil and fully functional
- KIdleTime support
- Plasma Surface - allows to tag windows as panels or notifications, allows absolute positioning
- Plasma WindowManagement - allows to get notifications about Windows. Used by the new libtaskmanager which we'll get in Plasma 5.7. Also hooked up in kwayland-integration repo for KWindowSystem. So you get e.g. windowAdded, windowActivated and a few more. If your taskmanager is based on KWindowSystem we can make sure that everything works without touching the code ;-) If you have a model: we expose a convenience model.

This should allow to have a quite decent experience on Wayland quickly. In fact I thought about just taking your source code and hack a day on it to have it fully migrated, but I'm just too overworked to find a time slot for it. The integration to get Plasmashell working went really fast and without problems. Task manager was quite some work as it's too X11 specific but other things were extremely simple.

Of course we also have other advantages. Wayland windows and X11 windows are considered the same. Features we have for X11 we also have for Wayland (if not they are being worked on). Giving you a good window management experience - KWin clearly has a better window management experience than Weston. We are doing that for 17 years now, we have quite some experience and covered all the corner cases.

We integrated KScreenLocker into KWin which provides a well established and secure lock screen infrastructure which can be styled with QtQuick through a Look'n'Feel package. We integrated KGlobalAccel, so that one has global shortcuts. Just recently we integrated QtVirtualKeyboard directly into KWin - again styleable through QtQuick.

Overall I think it really depends on how much manpower you have and how much you are willing to differ in feature between X11 and Wayland. For a smooth transition I think having identical feature set is an advantage. If you have many users already using KWin on X11, it might be best from user perspective to use KWin on Wayland. If you have many users screaming "Oh no KDE, the bloat" I would suggest to not use KWin.

Using KWin does not mean that you have to always use KWin. We really believe in modularity. Our protocols could be implemented by other compositors and if you use KWayland on the client side you can easily make that an optional runtime choice.

**Comment by paulolieuthier** [comment]
@mgraesslin 

> Those kinds of protocols will never be part of core protocol. They are needed for desktop environment, so if desktop environments want to see a standard arise they should come together and make it so.

> As @giucam mentioned we don't expect that this will ever go into Wayland core or even wayland-protocols.

Oh, I thought that's what xdg-shell was meant for. Looks like it aims much less than I thought.

> We think that the possibility to integrate multiple shells but also to exchange compositors is also important in a Wayland world.

We think that too. I believe it's key to LXDE and LXQt to be window manager-independent.

> KWin can shine in that regard as it makes a multi-process architecture and has many protocols already defined for that.

That's great. And it's awesome to know that we'll get a lot closer to feature parity by just continue using KWindowSystem. I have a question about that: if we stick to KWin, will we have to use wayland-specific code (KWayland) sometime or all features will be offered through kwayland-integration plugins? I mean, can we use **only** the display server-independent tier 1 libraries (KWindowSystem, KIdleTime etc.) and work on Wayland as well?

> In fact I thought about just taking your source code and hack a day on it to have it fully migrated, but I'm just too overworked to find a time slot for it.

That would have been cool. But we are already grateful for you and all KDE team's work on the Frameworks libraries we use. Also for your availability on email, irc and here on Github.

> If you have many users already using KWin on X11, it might be best from user perspective to use KWin on Wayland. If you have many users screaming "Oh no KDE, the bloat" I would suggest to not use KWin.

That's a very important discussion IMO. I've always used KWin (actually, I've never used LXDE, and I entered LXQt team after the merge) and I find Openbox, for example, not enoughly polished to me. However, AFAIK, Openbox is much more common for our users (someone please correct if I'm wrong) and therefore that needs to be discussed (I know Openbox is X11-only, I mean that people will want similarly lightweight Wayland compositors to replace it).

> Using KWin does not mean that you have to always use KWin. We really believe in modularity.

Well, it does look to me that on Wayland we will have to choose a window manager (protocol) to be tightly coupled to, unless we are willing to provide glue code for each protocol, which is complexity and resources I believe we can't afford. Ideally, a set of protocols like yours would be standardized and exchanging compositors would be trivial.

I really admire your work. Keep it up!

**Comment by mgraesslin** [comment]
> Oh, I thought that's what xdg-shell was meant for. Looks like it aims much less than I thought.

No, xdg-shell is basically just about how to create an application window. The name "shell" has nothing to do with a desktop shell.

> I have a question about that: if we stick to KWin, will we have to use wayland-specific code (KWayland) sometime or all features will be offered through kwayland-integration plugins?

Difficult to say. In Plasma we have both X11 and Wayland specific code. I'm not sure how much of that could be integrated directly through KWindowSystem - we did most of the porting before we had figured out how to make use of KWayland in KWindowsystem. Give me a list of the API methods you use and I can check ;-)

>  I mean, can we use only the display server-independent tier 1 libraries (KWindowSystem, KIdleTime etc.) and work on Wayland as well?

Just saying: kwayland-integration (which is currently not yet in frameworks, but I want to move it there) will be at least tier2 given that it needs to depend on KWayland+KWindowsystem and KWayland+KIdleTime respectively.

Apart from that: yes, mostly I think that will be possible. If you have X specific code somewhere (e.g. using Xlib or xcb directly or operating on X11 assumptions) that will need to be adjusted some why or the other.

> I mean that people will want similarly lightweight Wayland compositors to replace it

To a certain degree I think that will just not exist. If it's lightweight in the sense of lacking important features, one will just not have fun with such a compositor. It would mean you need to recreate lots of things it doesn't provide and in the end you will be at least as heavy-weight as the heavy-weight solution.

> Ideally, a set of protocols like yours would be standardized and exchanging compositors would be trivial.

We hope that our protocols could become kind of a de-facto standard here. Even if they are not added to Wayland proper if we have a few DEs with shared interests in that regard, we are all for standardizing them.

**Comment by paulolieuthier** [comment]
> Give me a list of the API methods you use and I can check ;-)

@mgraesslin I miss a way to request the window manager to close a window (or maybe I'm not seeing it). See here: https://github.com/lxde/lxqt-panel/blob/master/plugin-taskbar/lxqttaskbutton.cpp#L413.

**Comment by mgraesslin** [comment]
> I miss a way to request the window manager to close a window (or maybe I'm not seeing it). See here: https://github.com/lxde/lxqt-panel/blob/master/plugin-taskbar/lxqttaskbutton.cpp#L413.

Right that's using X11 specific code there (anything NET* is X11 specific). From a quick check it looks like we don't have a wrapper in KWindowSystem for that. The Wayland API replacement is https://api.kde.org/frameworks/kwayland/html/classKWayland_1_1Client_1_1PlasmaWindow.html#af387fd99c100dfa9d4a29db2e0978f9f

**Comment by paulolieuthier** [comment]
There should be a function for that in KWindowSystem and an implementation in KWayland integration plugin, right? I'll work on some patches.

**Comment by mgraesslin** [comment]
> There should be a function for that in KWindowSystem and an implementation in KWayland integration plugin, right?

Assuming there was no good reason to not have it: yes.

> I'll work on some patches.

Cool, thanks

**Comment by stqn1** [comment]
If I may give my opinion, since I just stumbled on this discussion… Recommending Xfwm which uses Gtk in a project named LXQt makes little sense to me. The reason I am trying LXQt is to get away from Gtk(3) and thus from Xfce. I have been happy with OpenBox so far; it just works and is as configurable as Xfwm.

Compton has been my compositor of choice since several years; it’s the best solution I’ve found to remove tearing and as a bonus I get a pretty desktop at the same time. The compositor of Xfwm 4.12 is not as good (it doesn’t remove tearing even when I enable the option and doesn’t have nice fade in/out options).

**Comment by agaida** [comment]
@stqn1 - thats easy - if you are happy with openbox and compton - use it. One could find fluxbox a good alternative too - some people use i3, some xfwm4, some kwin. So the Window Manager is a matter of taste, nothing less, nothing more.

And i agree to some extend that kwin will be the best choice, if it would have less dependencies or we would have the infrastructure around this WM in LXQt

**Comment by tsujan** [comment]
You have right to not believe me when I say [Compiz](https://github.com/compiz-reloaded) works much better than kwin and openbox+compton here.

After upgrading to kwin-5.10.3.1-1 on this new laptop, I encountered delay problems with kwin effects. compton wasn't an option due to its lacking elementary effects. So, I tried Compiz again: no delay, no lagging, no problem.

I wonder if it's really hard to make a decent compositing manager for X11? If yes, why was Compiz such a success years ago? Were its creators exceptional geniuses? ;) Or is it the case that there's so much focus on wayland that X11 is forgotten? Then this question arises: Why is wayland so elementary and have delay, lagging and many problems with a very new Intel card?

**Comment by simoniz0r** [comment]
Please don't.  I am very much enjoying LXQT with openbox and compton.  I do not enjoy xfwm4 much at all.  xfwm4 has some very, very bad screen tearing issues that compton does not have.  openbox is also much more configurable than xfwm4 is.

**Comment by tsujan** [comment]
@simoniz0r By "migrating away from OpenBox", it's only meant that OpenBox might not be recommended for LXQt in future; that's not decided yet -- for now, OpenBox *is* recommended. However, LXQt works and will work with OpenBox, KWin, xfwm4, Compiz,... or any other window manager that's modular enough and works fine. So, there's no need to worry.

**Comment by simoniz0r** [comment]
Alright, cool lol.  As long as openbox stays an option, I'm good.

**Comment by MttCastelli** [comment]
Hi all,
I can not follow you so deep in a tech discussion over WM and display server protocol but, can you explain me this? Am I using wayland over openbox or not?



[suse_laxqt_wayland_session.txt](https://github.com/lxde/lxqt/files/1389010/suse_laxqt_wayland_session.txt)

**Comment by stefonarch** [comment]
Openbox cannot be used in wayland. For some reason you have gdm processes, IMHO gdm is not a good choice as display manager for LXQt, it starts several gnome-related stuff.

`echo $XDG_SESSION_TYPE
`

tells you the session type.

**Comment by maverick74** [comment]
While @jleclanche might be right about Sway, i think it's worth bringing attention (in case anyone missed it) that there is now a collaboration between KDE and Sway that might make things a bit more interesting to LXQt 
(more info at: https://drewdevault.com/2018/04/28/KDE-Sprint-retrospective.html )

**Comment by agaida** [comment]
@maverick74 - Sway is nice for people in need of a modern tiling WM - thats right.

**Comment by maverick74** [comment]
@agaida My idea, when i read the post, was that there might be hope for sway as another "lite" alternative to openbox, in case we need it in LXQt.
I understand it's not as good as kwin, but maybe, for some, it will be enough :)

**Comment by tsujan** [comment]
Why continuing this discussion? Any DE-independent WM can be sued with LXQt. At some point, we'll have to remove `obconf-qt`, willy-nilly, and there will be no reason to make a Qt app for another WM.

**Comment by maverick74** [comment]
@tsujan but won't there be somekind of "recommended" WM when OB is removed?

(i didn't understand what you meant with "there will be no reason to make a Qt app for another WM"...)

**Comment by tsujan** [comment]
@maverick74 I'm sorry. You're right; I remembered that afterward and wanted to reply to my own comment. Apart from a simple recommendation, we should load a WM by default.

> i didn't understand what you meant with "there will be no reason to make a Qt app for another WM"

Like obconf-qt for openbox.

openbox and compton have no future -- as @agaida told in 2016, we should migrate away from them. A tiling WM can't be recommended by LXQt simply because most users prefer a free-floating one. kwin has some drawbacks. We can't depend on xfwm4, IMO... So, contrary to my last comment, this seems to need more discussion.

The best choice that comes to my mind is kwin, although I neither like nor use it.

**Comment by maverick74** [comment]
I use kwin with plasma a lot but, while i have no complains about it, i'm only using it _with_ plasma. I didn't test it on LXQt (where i'm using OB!).
I've read somewhere it could "slowdown"  LXQt a bit (which is not good for old machines), being that the reason o brought the Sway post to the discussion. It's supposed to be lite...

What are the _cons_ of Sway?

**Comment by simoniz0r** [comment]
kwin is not a good option, honestly.  You need KDE's settings app to be able to customize it at all.

There really is no *good* option for LXQt here.  The lightest option might be something like marco (which also requires a settings center app to be installed to work with, but all decent WMs are going to).

This is the kind of spot where  plugging in someone else's thing makes you not *really* feel like a DE at all.  You kinda just feel like a loose collection of tools that kind of work alright together.  *Maybe* if you designed your own settings application for kwin it could feel pretty good, but without any config or needing KDE's settings app installed to configure it, it's not really going to be a good tine for users.

**Comment by tsujan** [comment]
> kwin is not a good option, honestly. You need KDE's settings app to be able to customize it at all.

That's right but it's best among all.

> There really is no good option for LXQt here.

Also right.

> This is the kind of spot where plugging in someone else's thing makes you not really feel like a DE at all. 

I agree, unfortunately ;) IMO, that's a fact and it's better not to justify it by pointing to LXQt's modularity, which is another fact but a good one. We need a programmer who works in that area and wants to join LXQt. Or one of us might become supper active and do the job (not so probable). Until then, we have to be practical and rely on a WM. KWin is the best we can find for now.

> Maybe if you designed your own settings application for kwin it could feel pretty good

That may not be possible. KDE devs go in another direction. It's not unlikely that someday, kwin will become too dependent on KDE and its QML stuff, so that it couldn't be used outside it.

**Comment by maverick74** [comment]
>  but I'll surely try Sway to know it first hand. Your comment was an incentive.

@tsujan  Great to know that :D 
Please let us know about your experience with it.

@palinek tested Orbital, but i guess it was not a terribly good experience... Has that seen any improvement lately that could be a benefit for us?

> If, someday, LXQt starts to use it, I'll say goodbye to it

What's up with QML? Too slow? Resource hog? Limited?

**Comment by agaida** [comment]
Translate QML as synonym for PITA :smile: 

The only problem i see with sway is that user will run away screeming - those like me who prefer a "traditional" non-tiling WM. For users who want a tiling WM it can be a very good choice.

**Comment by tsujan** [comment]
> What's up with QML? Too slow? Resource hog? Limited?

In https://github.com/lxqt/lxqt/issues/1433 , I tried to explain an aspect of the problem in a simple language. There are others too (like JavaScript overhead and lack of flexibility). In short, (1) I don't want a discrepancy in look-and-feel of Qt and (2) I hate to write (and won't write) JS code to work with something that's directly available with C++.

**Comment by agaida** [comment]
And btw - the only thing i use QML for is for SDDM themes - even the debugging and formatting of the code written by others was painful enough so i can fully understand @tsujan

**Comment by simoniz0r** [comment]
> the only thing needed for configuration is kcmshell5

That's not even a package here.  `zypper` tells me it's part of `kde-cli-tools5` which doesn't exactly sound like a settings app?

**Comment by BabylonAS** [comment]
Regarding the tiling or non-tiling WM problem: over all the time I have used non-tiling window managers and understood its advantages over tiling WMs, mainly when I have to work with multiple applications at once. Also, a tiling WM may be cumbersome for computers with small resolution displays. While it is the end user’s freedom to use a tiling window manager, the new _default_ WM should keep the non-tiling paradigm as `openbox` does.

**Comment by maverick74** [comment]
LOL, I've heard lot's of people complaining about QML but never fully understand why... now i know. LOL
(this reminds me that there are truths better left unspoken ahahah)

@agaida like i mentioned above it _seems_ sway also supports floating and not just tiling - (take this with a LOT of salt because i don't know much about it :p )

**Comment by tsujan** [comment]
>  it seems sway also supports floating and not just tiling

The comment you mentioned implies that its floating support is still terrible ;) Maybe we should wait. However, although I use compiz-reloaded and can't compare anything with it, I also like to try a good tiling WM, which may be Sway.

**Comment by agaida** [comment]
@maverick74 - that would be great (or a good thing thing to start with). I've tested much WMs in the last few years - and the winner for me was xfwm4 right now - regarding speed, capabilities, configurations and memory usage.

**Comment by tsujan** [comment]
@agaida You and me don't use kwin but recommend it. So, kwin should be something, after all ;)

**Comment by BabylonAS** [comment]
@agaida But `xfwm4` is part of the Xfce project. Are we going to depend on even more other desktop environments (KDE comes to mind), or is it envisioned as a temporary solution until someone invents something better?

**Comment by maverick74** [comment]
@simoniz0r isn't it the systemsettings5 package?

@tsujan comment is 3 months old... maybe by some BIG miracle things got a _little_ improved ahah

**Comment by tsujan** [comment]
@maverick74 As far as I know, its author has been quite active. Let's hope he'll make a decent DE independent WM we could recommend.

**Comment by maverick74** [comment]
personally, i like kwin. And if that's the way i'm ok with it. I'm just concerned about possible bloat/resources misused.

but still, it's better than depending on gtk-based wm, imo.

i'm quite curios about your test results on sway as a floating wm @tsujan

**Comment by agaida** [comment]
@BabylonAS + @tsujan - the answer is simple, so i let the debian packaging speak:
```
xfwm4 | x-window-manager
```
i choose xfwm4 as default alternative or any package that provide the virtual package x-window-manager. Thats all - there is no hard dependency and there should be no hard dependency to a window manager. For debian i right now only strongly recommend xfmw4 (@tsujan mentioned the openbox and compton situation) - but in the end one is free to use what ever  one wants.

**Comment by tsloughter** [comment]
Instead of sway you might want to look at wlroots, the compositor library written for sway, https://github.com/swaywm/wlroots 

It is also being used by https://github.com/way-cooler/way-cooler and https://github.com/waymonad/waymonad

So wlroots could be used to build lxqt.

**Comment by agaida** [comment]
nice

**Comment by simoniz0r** [comment]
sway and such are cool, but Wayland is honestly a long ways off.

**Comment by agaida** [comment]
also true - but we had reduced the hard dependencies to X - so it is nice to have something to play with until LXQt is really ready for Wayland. You know, one should be prepared ...

**Comment by tsujan** [comment]
Another conclusion of mine: If we want our users to have something that really works instead of experimenting with this or that program, we could recommend KWin or xfwm4, IMO.

**Comment by simoniz0r** [comment]
Nothing wayland is ready for general use and won't be until they solve simple things like screen capturing not working easily for all applications.

IMO, much better to focus on where LXQt will be used now and figure out some nice ways to make it actually feel like a full on DE and not just a lose collection of tools.

**Comment by tsujan** [comment]
> Nothing wayland is ready for general use

Gnome-shell is. I'm saying this while I dislike it.

> and not just a lose collection of tools.

That's not fair. A good file manager that also handles desktop, a good panel, a practical set of configuration tools, a nice file dialog, etc. can't be called just a lose collection of tools. We're speaking about a WM here; nothing more.

**Comment by maverick74** [comment]
so, the conclusion on this bug (for now) is **Kwin or Xfwm4**... =)
(i had some hopes on Sway because of "liteness", but if it has that amount of bugs it's really better to leave it to "Timon and Pumbaa" - who really love bugs!)

While i do not agree that LXQt is "just a lose collection of tools" i've seen this image in a lot of places already and i agree with @simoniz0r we should focus a bit on make it feel a full DE or at least try to let people understand that it's modularity is a strong point, not a weakness :)

**Comment by tsujan** [comment]
Another word about things like Sway: We want something that works under an LXQt session, not something that needs its own session to work.

**Comment by tsujan** [comment]
@simoniz0r Who defended the infamous Gnome?! I meant its wayland support is the best when compared with KDE's and E's. And by lack of bugs I meant lack of crashes I saw under KDE and E. I can't tolerate gnome-shell more than 15 min. -- for the simplest actions I should do acrobatics with mouse or touchpad; it doesn't have a simple systray (once again, Gnome devs removed something they weren't able to handle); and....

**Comment by simoniz0r** [comment]
I wasn't talking about GNOME at all there really; I meant Wayland in general.  The things that I mentioned above are things that do not work well in any Wayland session.

**Comment by tsujan** [comment]
@simoniz0r I'm not a wayland fan. It's so shaky that when no crash happens, I say to myself, "they may really have done something this time..."

**Comment by tsujan** [comment]
@maverick74 At last, I succeeded in starting Sway. The secret was:

(1) Don't trust SDDM for that but login at a getty (Ctrl+Alt+F2 from SDDM screen, for me); and
(2) For me, the command was:
```
WLC_DRM_DEVICE=card1 sway
```
WLC isn't able to find the active card by itself!

Floating windows work fine. Actually, everything seems to work fine. However, to me, Sway is just like a game: It's fun to use it and I'll experiment with it in my free time but, apparently, I'm not the "tiling type" ;) It's also looks too elementary to me. Really? Remembering so many keyboard shortcuts to work with windows? Not for me and, I'm sure, not for many users but, again, it's fun ;)

BTW, I found a weird behavior in Sway's GitHub issue tracker: Sway devs like to close issues without clear explanations as soon as possible, as if they presuppose all users should know about its "obvious" secrets ;)

**Comment by maverick74** [comment]
@tsujan So,summing it up, it appears Sway is definitely a no-go for LXQt!

(I agree with you!!! I personally also value a good looking GUI =). I would not be in LXQt if it didn't provide one!!!)

Thank you for your time and tests!

**Comment by tsujan** [comment]
> t appears Sway is definitely a no-go for LXQt!

IMO, yes -- and not just because it's so ugly. I think @agaida agrees with that too.

Some people say that a good looking GUI distracts them from serious works. Some other consider ugly GUIs more "professional" and elegant ones "unprofessional". I enjoy pressing an elegant button inside a beautiful window while coding. Wayland has a long way to go before having the elegance X11 can provide -- if their developers value such things at all. Actually, wayland has a long way to go before it can be considered practical, IMO.

**Comment by maverick74** [comment]
@tsujan  I think professionalism has nothing to do with being beautiful or not. Neither with distractions.
But then again, some people still argue the earth is flat, so...

Do you (or anyone) have an idea about the resource/memory usage differences between OpenBox vs Kwin? Is is "that much  significant"?

**Comment by agaida** [comment]
@maverick74 - as all things - it depends how do you measure it. A wild guess it that kwin and friends have a basic mem footprint that is appox. 400 M higher as the same system with openbox - and that don't have to be a bad thing, if one have the RAM to don't care about.

Might sound nuts, but isn't - if i load the needed (and used libs) into RAM and hold them there it is a fine thing - i don't have to reload the things that i really use and that speed up all the things - and Qt does this excellent. So my suggestion would be: just try it with a typical workload and have a look at the memory usage after an hour of work.

**Comment by BabylonAS** [comment]
@agaida LXQt is envisioned as a `lightweight` desktop environment. While my ancient laptop using LXQt does have 2 gigabytes of RAM (which is much more than it’s supposed to have, if I recall correctly), it is still not much by current standards, and I would like to preserve as much free RAM space as possible. Also don't forget about small devices.

**Comment by tsujan** [comment]
@maverick74 I agree with @agaida. It depends on many things. Here, the difference is ~200 MiB at first but may grow to 400 after hours. On my current machine, with 16 GiB of RAM, it's negligible. On my old machine, with 4 GiB of RAM, it's something but not very important.

Here, with Compiz, the memory usage is about 650 MiB after boot + logging into LXQt. With KWin, it was about 700 MiB. With Openbox, about 500 MiB. These numbers depend on my settings and startup apps (Octopi, Copyq, CMST, Solaar,...), of course.

**Comment by simoniz0r** [comment]
I played around with kwin a bit last night with kcmshell5... I was not very happy with how it worked, tbh.  It would seem as if the entire KDE systemsettings5 app would need to be installed as I could not find any way to customize the hotkeys for interacting with windows without it.

**Comment by agaida** [comment]
@simoniz0r - only a wild guess - but you installed only the needed packages an leave Recommends out?

**Comment by agaida** [comment]
DISCLAIMER: kwin was developed for KDE and will have a bit of overhead - not our fault, not their fault as kwin wasn't developed for LXQt and is highly integrated into KDE

**Comment by simoniz0r** [comment]
Yes, but I did not see anything in recommended packages that provided that.

**Comment by maverick74** [comment]
@tsujan so, if LXQt with OB uses (for example) 150mb it would mean that with kwin it would use ~150mb - 200mb more.
Naturally this might not be nothing in certain systems, but for old PC's - to which lxqt is developed - i could indeed be very significant. (systems with 1gb of ram would be dead if one opens FF ahaha)

this is not good news!

**Comment by agaida** [comment]
@maverick74 - who said that LXQt is developed only for ancient machines short after WWII? - I runs fine on the oldes iron i have available and even this ancient machine (Centrino/Pentium M 32bit has 2G - standard was 1G at this time)

**Comment by maverick74** [comment]
@agaida ok ok... my fault!!! I apologize!!! Its developed _also_ for old machines LOL.

hummmmmm i wonder how well would it run on my good old ZX Spectrum? Anyone has an ISO ready for a tape???

(Oh... and i just stick it in a P4 with 1.5G)

**Comment by tsujan** [comment]
> but for old PC's - to which lxqt is developed

LXQt isn't being developed *specifically* for old machines -- as @agaida said. One of the goals is keeping it lightweight *as far as possible* but that doesn't mean sacrificing functionality for that. My Laptop is very powerful and still I use and work on LXQt with it.

**Comment by agaida** [comment]
ok, some numbers - my current workhorse
```System:    Host: archramme Kernel: 4.16.8-1-ARCH x86_64 bits: 64 Desktop: LXQT 0.12.0 Distro: Arch Linux 
CPU:       Topology: 8-Core model: AMD FX-8350 bits: 64 type: MCP L2 cache: 2048 KiB 
           Speed: 1665 MHz min/max: 1400/4000 MHz Core speeds (MHz): 1: 1666 2: 2423 3: 1512 4: 1517 5: 1957 
           6: 3794 7: 1588 8: 3233 
Graphics:  Card-1: NVIDIA GM206 [GeForce GTX 950] driver: nvidia v: 396.24 
           Display: x11 server: X.Org 1.19.6 driver: nvidia unloaded: fbdev,modesetting,nouveau,nv,vesa 
           resolution: 1920x1200~60Hz 
           OpenGL: renderer: GeForce GTX 950/PCIe/SSE2 v: 4.6.0 NVIDIA 396.24 
Info:      Processes: 204 Uptime: 20h 22m Memory: 31.32 GiB used: 1.98 GiB (6.3%) Shell: zsh inxi: 3.0.07 
```
my highspeed pentium m
```
just a second please
```

**Comment by BabylonAS** [comment]
@agaida What an interesting coincidence (my laptop also uses Intel Pentium M, and I have already mentioned the RAM volume as being 2 GiB). But someone might be in a position when a fifteen-years-old computer is the only one he or she could use for some time — and I am going to. While I usually don’t run a lot of intensive programs on my laptop (Firefox is surely the heaviest, although I could pretty well stick to Opera if only it would be able to run Discord web client properly), at some point it may be cumbersome to handle. Surely it is possible to tolerate lags from such intensive activity, but there always might be something to improve.

**Comment by agaida** [comment]
and to be honest - and with a bit of sarcasm - as the well know psychologist Mick Jagger said years ago: You can't have everything ...

So it is up to the distributions to set up their distributions regarding to their goals - and i guess no one @ antix or so would ever mind to deliver LXQt with kwin - but thats FOSS, one can mix things as one want.

**Comment by agaida** [comment]
updated the WM Post with openbox and kwin

**Comment by maverick74** [comment]
@agaida WOW!!! Really surprised about **Kwin** as well!!! **_Just 30Mb over OpenBox_**

That will do for me!!! Kwin it is!!!

**Comment by agaida** [comment]
thats again no problem of the Window Manager and don't fit in here ...

**Comment by simoniz0r** [comment]
How is it not?  A user is going to want to, for example, be able to change the hotkey of how to close windows.  I could not find a way to do that without systemsettings5 for kwin.

**Comment by maverick74** [comment]
Can't, **theoretically speaking**, LXQt's current shortcut manager detect if one is using kwin and in that case work the way kwin needs?

**Comment by agaida** [comment]
Added a test with mutter - @tsujan - what about gnome-shell and mutter as default for wayland ...

/me runs away screeming

**Comment by maverick74** [comment]
Completely amazed with Mutter!!! I never thought it would beat kwin!!! Amazing!!!
@agaida what do you mean with:
> did i mention the great Qt sharing of ressources?


can you clarify?

**Comment by simoniz0r** [comment]
@agaida I would be fine with kwin if there were a way to set hotkeys such as closing windows without needing systemsettings5.  In my tests, if I disabled kwin's compositor and used compton instead, kwin was actually pretty light on RAM use.  It was using ~20MBs of RAM without compositing and ~70MBs with.

**Comment by agaida** [comment]
@maverick74 - the most qt things come with a relative hight memory footprint at the beginning - notable more than in GTK envionments. But if you really use the system the memory usage don't grow so much as one would expect without the knowledge needed - so if you have a look at LXQt, we use kwindowsystem (Tier 1 KF5 component), kidletime (Tier 1), solid (Tier 1) and what ever i forget - so lxqt load the needed part into memory and with the base set of LXQt components the very most basic Qt libs a Qt application will need is already loaded and can be re-used. That means - the goal is not to save memory at any price - the goal should be to use the available memory wise. And in case even a "fat" thing like kwin don't add much to the overall memory footprint. And thats really cool.

**Comment by maverick74** [comment]
@agaida I didn't know about that. That's a very clever approach, imo!!!
Thanks for bringing some light on the subject :)

**Comment by tsujan** [comment]
> Added a test with mutter - @tsujan - what about gnome-shell and mutter as default for wayland ...

I'm working under it now, with pcmanfm-qt, qterminal, featherpad,... and without Gnome apps. Everything is fine.

@simoniz0r

As for KDE shortcuts under LXQt, they shouldn't work, of course, unless KGlobalAccel is running, which isn't a good idea outside KDE. I used KWin with LXQt without any problem for a long time and changed kwin themes with systemsettings5. Whenever I ran systemsettings5, some KDE processes started and I stopped those which were irrelevant.

**Comment by tsujan** [comment]
Of course, I was working under gnome-shell just to test Qt 5.11 -- which was a a bad experience. Returning to LXQt.....

**Comment by agaida** [comment]
@maverick74 - and i'm not exactly sure that i'm 100% right about the usage thing - but if not someone will step up and corret me :D

**Comment by tsujan** [comment]
> I would suspect it uses gnome's windows-registry-wannabe for configuration, is it not?

In case you asked me, I think gnome should use gnome stuff, of course. Qt apps use xwayland under it. I didn't encounter a single problem regarding window management with gnome-shell+wayland+Intel.

**Comment by Vladimir-csp** [comment]
I mean dconf or gsettings or whatever iteration of this monstrosity they have now.

**Comment by tsujan** [comment]
It should be dconf.

**Comment by maverick74** [comment]
So... all in all, from what i can understand, if one has at least 1G of ram one can use whatever wm he wants because the differences are not really that significant. (except for others details like most stable/complete/advanced).

**Comment by agaida** [comment]
yes, that was the plan right from the beginning

**Comment by maverick74** [comment]
yes, i believe it was! But i wasn't really expecting such good results to be honest!!!
I was expecting big differences from wm to wm

**Comment by maverick74** [comment]
A _suggestion_: Picking up @PCMan 's [blog post](https://blog.lxde.org/2016/10/04/benchmark-memory-usage-lxqt-desktop-environment-vs-xfce/) as an example, i think this too would give a very interesting articlet!!! 
I bet other users would also like to know about this experiments (that, at the same time would also put an end to some myths/fears/misconceptions )

**Comment by agaida** [comment]
we should do such things more often - right. But it's hard for me as not natural english speaker to put together such an article - so if one will stand up and do this ...

But the: It just works is only one side of the coin - the second side is also important - as seen in the discussion. How to integrate the WM (or any other external application) into LXQt - fortunately not every LXQt-User is also a LXQt-Developer, that would be a bit boring and frustrating :D

**Comment by tsujan** [comment]
I've seen this in other places but have no idea whether it's because of Linux kernel, C++ or Qt: A Qt program may take, let's say, 100 MiB of RAM when the total RAM is 16 GiB but the same program may take only 20 MiB with a 4 GiB RAM (the numbers are totally hypothetical). Until now, I haven't been so curious (or have been too lazy) to search the Internet for it.

**Comment by maverick74** [comment]
@agaida I would gladly step up! But English is not my native language either...

I think the capacity  of integrating various WM (or any other external application) is a plus! Even if it is not for the faint of heart, it's still a plus.
In the article, i believe we would not need to get into the technical side of it, but mostly share the results so that a reader could know of the actual state and capacities of LXQt regarding the use of different WM's

**Comment by chimak111** [comment]
@agaida , @maverick74 
I think many English speakers and readers aren't native. Even the "natives" are quite understanding and forgiving. So please don't hesitate!

**Comment by tsujan** [comment]
I played with the code of KWin's Breeze titlebar -- starting from [BreezeBlur](https://github.com/alex47/BreezeBlurred) because it didn't have a different config file -- and saw that the font can be configured outside KDE. This is after heavy and, sometimes, dirty hacks (mac-like titlbar buttons grow on mouse-over):

![breeze](https://user-images.githubusercontent.com/981076/46264841-f803d100-c52d-11e8-84fd-9b206d19aedb.png)

Of course, it depends on KDE but is completely configurable under LXQt.

No upload yet because it needs a lot of cleanup and I should see whether KDE breaks the Breeze code once in a while (I don't think so).

**Comment by shirishag75** [comment]
Here are my numbers - 

```
$ inxi -SICGxxx
System:    Host: debian Kernel: 4.18.0-2-amd64 x86_64 bits: 64 compiler: gcc v: 7.3.0 Desktop: LXQt 0.13.0 tk: Qt 5.11.1 
           info: lxqt-panel wm: xfwm4 dm: LightDM 1.26.0 Distro: Debian GNU/Linux buster/sid 
CPU:       Topology: Quad Core model: Intel Core i5-7400 bits: 64 type: MCP arch: Kaby Lake rev: 9 L2 cache: 6144 KiB 
           flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 24000 
           Speed: 800 MHz min/max: 800/3300 MHz Core speeds (MHz): 1: 800 2: 800 3: 800 4: 800 
Graphics:  Device-1: Intel HD Graphics 630 vendor: ASUSTeK driver: i915 v: kernel bus ID: 00:02.0 chip ID: 8086:5912 
           Display: x11 server: X.Org 1.20.1 driver: modesetting unloaded: fbdev,vesa resolution: 1600x900~60Hz 
           OpenGL: renderer: Mesa DRI Intel HD Graphics 630 (Kaby Lake GT2) v: 4.5 Mesa 18.1.7 compat-v: 3.0 
           direct render: Yes 
Info:      Processes: 181 Uptime: 9m Memory: 7.66 GiB used: 1.01 GiB (13.2%) Init: systemd v: 239 runlevel: 5 Compilers: 
           gcc: 8.2.0 alt: 8 Shell: bash v: 4.4.23 running in: qterminal inxi: 3.0.27
```

**Comment by glpzzz** [comment]
I'm having a very good experience using tiling wm with LxQt, so far I tried i3, bspwm and dwm. Of course, just for the apps, the session management, and other stuff because desktop, panel, runner, is not used anymore.

**Comment by redtide** [comment]
Someone recently suggested me this interesting project: <https://github.com/labwc/labwc>. I had already started trying something else, but maybe someone here could be interested, seems to be a very lightweight wayland stacking compositor using openbox 3.4 specification for configuration files.

**Comment by tsujan** [comment]
LXQt doesn't make an exception for openbox or compton anymore. Closing...
